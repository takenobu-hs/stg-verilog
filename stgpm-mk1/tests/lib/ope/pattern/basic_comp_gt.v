

`define SP1_COMP_GT

  parameter DW = 32;


  /* test pattern */

  initial begin
    @(negedge rst);
    repeat(2) @(posedge clk);

    @(posedge clk);
    a0 = 32'h0000_0002;
    a1 = 32'h0000_0002;

    @(posedge clk);
    a0 = 32'h0000_0001;
    a1 = 32'h0000_0002;

    @(posedge clk);
    a0 = 32'h0000_0002;
    a1 = 32'h0000_0001;


    @(posedge clk);
    a0 = 32'h0000_0000;
    a1 = 32'hffff_ffff;

    @(posedge clk);
    a0 = 32'hffff_ffff;
    a1 = 32'h0000_0000;

    @(posedge clk);
    a0 = 32'h0000_0000;
    a1 = 32'h0000_0000;

    @(posedge clk);
    a0 = 32'hffff_ffff;
    a1 = 32'hffff_ffff;

    @(posedge clk);
    a0 = 32'h8000_0000;
    a1 = 32'h8000_0000;

    @(posedge clk);
    a0 = 32'h8000_0000;
    a1 = 32'h7fff_ffff;

    @(posedge clk);
    a0 = 32'h7fff_ffff;
    a1 = 32'h8000_0000;

    repeat(5) @(posedge clk);
    halt();
  end


