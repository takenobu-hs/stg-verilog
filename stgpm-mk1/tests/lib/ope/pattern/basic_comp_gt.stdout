monitor   :       0: xxxxxxxx xxxxxxxx  x
monitor   :      80: 00000002 00000002  0
monitor   :      90: 00000001 00000002  0
monitor   :     100: 00000002 00000001  1
monitor   :     110: 00000000 ffffffff  0
monitor   :     120: ffffffff 00000000  1
monitor   :     130: 00000000 00000000  0
monitor   :     140: ffffffff ffffffff  0
monitor   :     150: 80000000 80000000  0
monitor   :     160: 80000000 7fffffff  1
monitor   :     170: 7fffffff 80000000  0
